samples/bluetooth/central_uart/src/main.c

```diff
@@ -355,12 +353,125 @@ static int scan_init(void)
 	return err;
 }
 
+static void gatt_discover(struct bt_conn *conn)
+{
+	if (conn == default_conn) {
+		int err = bt_gatt_dm_start(conn,
+					   BT_UUID_NUS_SERVICE,
+					   &discovery_cb,
+					   &gatt_nus_c);
+		if (err) {
+			printk("could not start the discovery procedure, error "
+			       "code: %d\n", err);
+		}
+	}
+}
+
+static void auth_passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Passkey for %s: %06u\n", addr, passkey);
+}
+
+static void auth_passkey_confirm(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	auth_conn = bt_conn_ref(conn);
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Passkey for %s: %06u\n", addr, passkey);
+	printk("Press Button 1 to confirm, Button 2 to reject.\n");
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+
+	gatt_discover(conn);
+}
+
+static void auth_done(struct bt_conn *conn)
+{
+	printk("%s()\n", __func__);
+	bt_conn_auth_pairing_confirm(conn);
+
+	gatt_discover(conn);
+}
+
+
+static void pairing_complete(struct bt_conn *conn, bool bonded)
+{
+	printk("Paired conn: %p, bonded: %d\n", conn, bonded);
+
+	gatt_discover(conn);
+}
+
+static void pairing_failed(struct bt_conn *conn)
+{
+	printk("Pairing failed conn: %p\n", conn);
+
+	gatt_discover(conn);
+}
+
+static struct bt_conn_auth_cb conn_auth_callbacks = {
+	.passkey_display = auth_passkey_display,
+	.passkey_confirm = auth_passkey_confirm,
+	.cancel = auth_cancel,
+	.pairing_confirm = auth_done,
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed
+};
+
+static void num_comp_reply(struct bt_conn *conn, bool accept)
+{
+	if (accept) {
+		bt_conn_auth_passkey_confirm(conn);
+		printk("Numeric Match, conn %p\n", conn);
+	} else {
+		bt_conn_auth_cancel(conn);
+		printk("Numeric Reject, conn %p\n", conn);
+	}
+}
+
+static void button_handler(u32_t button_state, u32_t has_changed)
+{
+	u32_t buttons = button_state & has_changed;
+
+	if (auth_conn) {
+		if (buttons & KEY_PASSKEY_ACCEPT) {
+			num_comp_reply(auth_conn, true);
+		}
+
+		if (buttons & KEY_PASSKEY_REJECT) {
+			num_comp_reply(auth_conn, false);
+		}
+
+		bt_conn_unref(auth_conn);
```

Comment:
To support more buttons you should move this into num_comp_reply

Reply:
Ok. Moved